sql-ballerina/Module.md

```diff
@@ -0,0 +1,364 @@
+                               connectionPool = { maxOpenConnections: 5 });
+    ```
+
+3. Local, shareable connection pool
+
+    If you create a record of type `sql:ConnectionPool` and reuse that in the configuration of multiple clients, 
+    for each  set of clients that connects to the same database instance with the same set of properties, a shared 
+    connection pool will be created. The JDBC module example below shows how the global connection pool is used.
+
+    ```ballerina
+    sql:ConnectionPool connPool = {maxOpenConnections: 5};
+    
+    jdbc:Client|sql:Error dbClient1 =       
+                               new (url = "jdbc:mysql://localhost:3306/testdb",
+                               connectionPool = connPool);
+    jdbc:Client|sql:Error dbClient2 = 
+                               new (url = "jdbc:mysql://localhost:3306/testdb",
+                               connectionPool = connPool);
+    jdbc:Client|sql:Error dbClient3 = 
+                               new (url = "jdbc:mysql://localhost:3306/testdb",
+                               connectionPool = connPool);
+    ```
+    
+#### Closing the Client
+
+Once all the database operations are performed, you can close the database client you have created by invoking the `close()`
+operation. This will close the corresponding connection pool if it is not shared by any other database clients. 
+
+```ballerina
+error? e = dbClient.close();
+```
+Or
+```ballerina
+check dbClient.close();
+```
+
+### Database Operations
+
+Once the client is created, database operations can be executed through that client. This module defines the interface 
+and common properties that are shared among multiple database clients.  It also supports querying, inserting, deleting, 
+updating, and batch updating data.  
+
+#### Creating Tables
+
+This sample creates a table with two columns. One column is of type `int`and the other is of type `varchar`.
+The CREATE statement is executed via the `execute` remote function of the client.
+
+```ballerina
+// Create the ‘Students’ table with the  ‘id’, 'name' and ‘age’ fields.
+sql:ExecutionResult|sql:Error ret = dbClient->execute("CREATE TABLE student(id INT AUTO_INCREMENT, " +
+                         "age INT, name VARCHAR(255), PRIMARY KEY (id))");
+if ret is sql:ExecutionResult {
+    //A value of type sql:ExecutionResult is returned for 'ret'. 
+    //Any operations can be performed using the variable 'ret' and fields in sql:ExecutionResult can be accessed.
+} else {
+    //an error of type sql:Error is returned for 'ret'.
+}
+```
+
+#### Inserting Data
+
+This sample shows three examples of data insertion by executing an INSERT statement using the `execute` remote function 
+of the client.
+
+In the first example, the query parameter values are passed directly into the query statement of the `execute` 
+remote function.
+
+```ballerina
+sql:ExecutionResult|sql:Error ret = dbClient->execute("INSERT INTO student(age, name) " +
```

Comment:
Shall we use check here? Check-in all places

Reply:
That is better! added checks in https://github.com/ballerina-platform/module-ballerina-sql/pull/228/commits/05117616859a5880c2b2742a3c249854feb8b20a